frontend/pages/hosts/ManageHostsPage/HostTableConfig.tsx

```diff
@@ -261,12 +270,54 @@ const defaultHiddenColumns = [
   "hardware_serial",
 ];
 
-const generateVisibleHostColumns = (
-  hiddenColumns: string[]
+/**
+ * Will generate a host table column configuration based off of the current user
+ * permissions and license tier of fleet they are on.
+ */
+const generateAvailableTableHeaders = (
+  config: IConfig,
+  currentUser: IUser
 ): IHostDataColumn[] => {
-  return hostTableHeaders.filter((column) => {
+  return allHostTableHeaders.reduce(
+    (columns: IHostDataColumn[], currentColumn: IHostDataColumn) => {
+      // skip over column headers that are not shown in core tier
+      if (
+        permissionUtils.isCoreTier(config) &&
+        (permissionUtils.isGlobalAdmin(currentUser) ||
+          permissionUtils.isGlobalMaintainer(currentUser))
+      ) {
+        if (
+          currentColumn.accessor === "team_name" ||
+          currentColumn.id === "selection"
+        ) {
+          return columns;
+        }
```

Comment:
Not sure if I'm reading this correctly, but I think `team_name` should still be an accessible column to any user on Fleet Basic. @noahtalerman can you please confirm?

Reply:
Correct. The `team_name` column should be presented to any user on Fleet Basic. This includes all Global and Team users.

Reply:
ah damn. yah my bad. i keep getting basic and core mixed up. will change

Reply:
i looked into this more. The showing of the team column seems to be right, but did need to make changes for showing the selection column